### PR TITLE
[SPARK-21856] Add probability and rawPrediction to MLPC for Python

### PR DESCRIPTION
**Pull request by chunshengji**
Probability and rawPrediction has been added to MultilayerPerceptronClassifier for Python

Add unit test.

### CODE REVIEW

**Comment by WeichenXu123** [comment]
Jenkins, test this please.

File: python/pyspark/ml/tests.py [hunk 1]
```diff
@@ -1655,6 +1655,25 @@ def test_multinomial_logistic_regression_with_bound(self):
             np.allclose(model.interceptVector.toArray(), [-0.9057, -1.1392, -0.0033], atol=1E-4))
 
 
+class MultilayerPerceptronClassifierTest(SparkSessionTestCase):
+
+    def test_multilayer_perceptron_classifier(self):
+
+        data_path = "data/mllib/sample_multiclass_classification_data.txt"
```
Comment (WeichenXu123) [inline python/pyspark/ml/tests.py]: I prefer data set generated in code. Like other test cases.

File: python/pyspark/ml/tests.py [hunk 2]
```diff
@@ -1655,6 +1655,25 @@ def test_multinomial_logistic_regression_with_bound(self):
...
+        df = self.spark.read.format("libsvm").load(data_path)
+
+        mlp = MultilayerPerceptronClassifier(maxIter=100, layers=[4, 5, 4, 3], blockSize=1, seed=123)
```
Comment (WeichenXu123) [inline python/pyspark/ml/tests.py]: We can simplify the layer structure to speed up tests. E.g, use 3 layers structure.

File: python/pyspark/ml/classification.py [hunk 3]
```diff
@@ -1425,11 +1425,13 @@ class MultilayerPerceptronClassifier(JavaEstimator, HasFeaturesCol, HasLabelCol,
...
                  maxIter=100, tol=1e-6, seed=None, layers=None, blockSize=128, stepSize=0.03,
-                 solver="l-bfgs", initialWeights=None):
+                 solver="l-bfgs", initialWeights=None, probabilityCol="probability",
+                 rawPredicitionCol="rawPrediction"):
         """
         __init__(self, featuresCol="features", labelCol="label", predictionCol="prediction", \
                  maxIter=100, tol=1e-6, seed=None, layers=None, blockSize=128, stepSize=0.03, \
-                 solver="l-bfgs", initialWeights=None)
+                 solver="l-bfgs", initialWeights=None, probabilityCol="probability", \
+                 rawPredicitionCol="rawPrediction")
```
Comment (WeichenXu123) [inline python/pyspark/ml/classification.py]: Also add `probabilityCol="probability", rawPredicitionCol="rawPrediction"` into the following line: 
```self._setDefault(maxIter=100, tol=1E-6, blockSize=128, stepSize=0.03, solver="l-bfgs")```

**Review by WeichenXu123 — COMMENTED** [review]
Thanks! I leave some comments.

File: python/pyspark/ml/classification.py (hunk 3 above)
Comment (yanboliang) [reply python/pyspark/ml/classification.py]: @WeichenXu123 Actually these default values has been set in base classes ```HasProbabilityCol``` and ```HasRawPredictionCol```, so we don't to set it again here.

File: python/pyspark/ml/classification.py [hunk 4]
```diff
@@ -1442,11 +1444,13 @@ def __init__(self, featuresCol="features", labelCol="label", predictionCol="pred
...
                   maxIter=100, tol=1e-6, seed=None, layers=None, blockSize=128, stepSize=0.03,
-                  solver="l-bfgs", initialWeights=None):
+                  solver="l-bfgs", initialWeights=None, probabilityCol="probability",
+                  rawPredicitionCol="rawPrediction"):
         """
         setParams(self, featuresCol="features", labelCol="label", predictionCol="prediction", \
                   maxIter=100, tol=1e-6, seed=None, layers=None, blockSize=128, stepSize=0.03, \
-                  solver="l-bfgs", initialWeights=None)
+                  solver="l-bfgs", initialWeights=None, probabilityCol="probability", \
+                 rawPredicitionCol="rawPrediction"):
```
Comment (yanboliang) [inline python/pyspark/ml/classification.py]: Align.

File: python/pyspark/ml/tests.py (hunk 1 above)
Comment (yanboliang) [reply python/pyspark/ml/tests.py]: For other algorithms, I agree with you. However, too simple dataset is not ideal to test MLPC. This dataset is a tiny dataset with 150 rows and 4 columns, so I think we can use it. Thanks.

**Comment by gatorsmile** [comment]
ok to test

File: python/pyspark/ml/tests.py [hunk 5]
```diff
@@ -1655,6 +1655,26 @@ def test_multinomial_logistic_regression_with_bound(self):
             np.allclose(model.interceptVector.toArray(), [-0.9057, -1.1392, -0.0033], atol=1E-4))
 
 
+class MultilayerPerceptronClassifierTest(SparkSessionTestCase):
+
+    def test_multilayer_perceptron_classifier(self):
```
Comment (yanboliang) [inline python/pyspark/ml/tests.py]: Rename to ```test_raw_and_probability_prediction```?

**Review by yanboliang — COMMENTED** [review]
One minor comment, otherwise, LGTM. Thanks.

**Review by WeichenXu123 — APPROVED** [review]
LGTM. Thanks!

**Comment by yanboliang** [comment]
LGTM2, merged into master. Thanks.

**Review by yanboliang — APPROVED** [review]
LGTM2, merged into master. Thanks.

**Pull request closed without merging**